frontend/web/src/app/components/material/material-detail-panel/material-detail-panel.component.ts

```diff
@@ -0,0 +1,136 @@
+import {Component, EventEmitter, Input, OnChanges, OnInit, Output} from "@angular/core";
+import {Material} from "../../../model/material";
+import {MaterialService} from "../../../services/material/material.service";
+import {Router} from "@angular/router";
+import {CustomerService} from "../../../services/customer/customer.service";
+
+declare var $: any;
+
+@Component({
+  selector: 'app-material-detail-panel',
+  templateUrl: './material-detail-panel.component.html',
+  styleUrls: ['./material-detail-panel.component.scss'],
+  providers: [MaterialService, CustomerService],
+})
+export class MaterialDetailPanelComponent implements OnInit, OnChanges {
+  private materialTable;
+  private propertyTable;
+  private loadingTable = false;
+  private dataLoaded = false;
+  @Input('material') material: Material;
+  @Output() closeReq: EventEmitter<null> = new EventEmitter<null>();
+  private dateString: string;
+  private materialPropertiesFieldNames;
+
+  constructor(private customerService: CustomerService,
+              private materialService: MaterialService,
+              private router: Router) {
+  }
+
+  ngOnInit() {
+    this.materialTable = $('#materialGradeTable');
+    this.propertyTable = $('#materialPropertyTable')
+    if (this.material) {
+      this.getMaterialInfo();
+    }
+  }
+
+  private getMaterialInfo() {
+    this.materialService.getMaterial(this.material.id).subscribe(
+      (material) => {
+        this.material = material;
+        this.loadTables();
+      },
+    );
+  }
+
+  ngOnChanges() {
+    this.loadTables();
+    if (this.material) {
+      this.getMaterialInfo();
+    }
+  }
+
+  private loadTables() {
+    if (!this.loadingTable && this.materialTable) {
+      this.loadingTable = true;
+      this.populateMaterialGrades();
+      this.populateMaterialProperties();
+      this.loadingTable = false;
+    }
+  }
+
+  private populateMaterialProperties() {
+    const _self = this;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assigning `this` reference to local variable not allowed: _self. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4327)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "_self". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1854)